modules/social_features/social_content_report/config/optional/language.content_settings.taxonomy_term.report_reasons.yml

```diff
@@ -0,0 +1,15 @@
+langcode: en
+status: true
+dependencies:
+  config:
+    - taxonomy.vocabulary.report_reasons
+  module:
+    - social_content_translation
+third_party_settings:
```

Comment:
Could you ellaborate why this is set to false and only enabled in the config override?

Reply:
Hi @ronaldtebrake !
This is a specific how translations work in Open Social. When OS become multilingual translations will not work until we enable the small module `social_content_translation` which loads all config overrides responsible for translating. In other words, until we enable this module all entities supports translating but aren't translatable. 
Why we don't set the field value `true` by default? Because this gives us opportunity to control entities translation on whole platform by enabling/disabling the module `social_content_translation` - if enabled entities are translatable, if disabled - they aren't.

Reply:
But you already have a dependency set to
```
  module:
    - social_content_translation
```

So this config will only apply once social content translation is enabled right?

Reply:
@ronaldtebrake Correct. But if we even enable `social_content_translation` later this config will be installed anyway (see [here](https://github.com/goalgorilla/open_social/blob/main/modules/custom/social_language/modules/social_content_translation/social_content_translation.install#L33)).

Reply:
Hi @nechai sorry totally missed your comment here.
What I'm trying to say is that, I believe the way `config/optional` works is that whenever you have dependencies added to your `config/optional` config file, in this case the dependencies being:

```
dependencies:
  config:
    - taxonomy.vocabulary.report_reasons
  module:
    - social_content_translation
```

This config will only be installed when all of the dependencies are met.

So I would imagine the enabled could be set to true immediately and we don't even need the config override for this. Because if the module isn't enabled, this config is ignored because not all the dependencies are met.

For more background info: https://www.drupal.org/node/2453919

Reply:
Hi @ronaldtebrake !
If any module is enabled **after** `social_content_translation` module then dependencies are met and config will be installed.
If any module is enabled **before** `social_content_translation` module then [the function](https://github.com/goalgorilla/open_social/blob/main/modules/custom/social_language/modules/social_content_translation/social_content_translation.install#L33) will do the trick and install every missed config.

Regarding having `enabled: true` key-value by default.
Imagine this case. We have a multilingual platform. Then we decided to move it back to single language. Now, if we disable the `social_content_translation` all configs overrides will be disabled and value will remain `enabled: false` and all our entities will become untranslatable. If we set `enabled: true` by default after disabling mentioned module entities remains translatable. This is just not a perfect way to achieve ability having `on/off` multi-language feature.

Reply:
Hi @nechai 

Thanks for the explanation, i'm still a bit confused, sorry about that.

> If any module is enabled before social_content_translation module then [the function](https://github.com/goalgorilla/open_social/blob/main/modules/custom/social_language/modules/social_content_translation/social_content_translation.install#L33) will do the trick and install every missed config. 

That code actually lives in `social_content_translation` and does exactly what `config/optional` should do for you.

Imagine, when someone enables `social_content_translation`, Drupal will go through all the modules and their `config/optional` files for you, and check if it needs to import any additional config based on if certain dependencies are met. at least that is what the documentation says when I read it here: https://www.drupal.org/node/2453919

> When a module is installed that provides a configuration entity type, for example Views, all installed modules' config/optional directories are searched for any configuration entities of the new type. If any configuration entities are found and their other dependencies can be met, the configuration entities are created. Given the original example of Views and Node, this means it does not matter which way around Views and Node are installed. If Node is installed first, the frontpage view is created when Views is installed. If Views is installed first, the frontpage view is created when Node is installed.

Or is this something we've tested and it doesn't work, in that case I totally see your point! (Sorry for the confusion here)

With regards to this:

> Imagine this case. We have a multilingual platform. Then we decided to move it back to single language. Now, if we disable the social_content_translation all configs overrides will be disabled and value will remain enabled: false and all our entities will become untranslatable. If we set enabled: true by default after disabling mentioned module entities remains translatable. This is just not a perfect way to achieve ability having on/off multi-language feature.

Isn't that something we have to worry about when we uninstall/disable the `social_content_translation`, so we write that in the `hook_uninstall`? 
So when you enable the module you set it to `enabled: true` through the default config, when you uninstall/disable you set it back to `enabled: false` through a `hook_uninstall`

Why I prefer not to have a ConfigOverridde because its something called every single time on runtime and noboody can change this anymore through the UI, so any Drupal open source user is stuck to that whenever the module is enabled, regardless if they want to switch it `on/off` themselves.